tracing-core/src/field.rs

```diff
@@ -194,6 +194,16 @@ pub trait Visit {
         self.record_debug(field, &value)
     }
 
+    /// Visit a signed 128-bit integer value.
+    fn record_i128(&mut self, field: &Field, value: i128) {
+        self.record_debug(field, &value)
+    }
+
+    /// Visit an unsigned 128-bit integer value.
+    fn record_u128(&mut self, field: &Field, value: u128) {
+        self.record_debug(field, &value)
+    }
```

Comment:
I think it's correct that these go straight to `record_debug`, since a narrowing conversion to `u64`/`i64` could silently lose data.

i _do_ kind of wonder if we would want `record_i64`/`record_u64` to default to trying to cast that value as a 128-bit integer before going straight to `record_debug` --- that way, the visitor would only need to implement one set of hooks to record any integer. but, on the other hand, that would mean passing a bigger value to an additional function...so i'm not sure if it's worth it. i think the current approach is fine for now, and we can always addressed this in a follow-up.

Reply:
> i do kind of wonder if we would want record_i64/record_u64 to default to trying to cast that value as a 128-bit integer before going straight to record_debug --- that way, the visitor would only need to implement one set of hooks to record any integer. but, on the other hand, that would mean passing a bigger value to an additional function...so i'm not sure if it's worth it. i think the current approach is fine for now, and we can always addressed this in a follow-up.

I _was_ thinking about this and you stated my concerns—I wouldn't be surprised if converting to a larger number had a negative performance impact, but yes, this could be good to benchmark.